File: VIPER Tests.xctemplate/Objective-C/___FILEBASENAME___PresenterTests.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import <UIKit/UIKit.h>
#import <XCTest/XCTest.h>

#import "___FILEBASENAMEASIDENTIFIER___Presenter.h"
#import "___FILEBASENAMEASIDENTIFIER___PresenterProtocols.h"

@interface ___FILEBASENAMEASIDENTIFIER___PresenterTests : XCTestCase <
        ___FILEBASENAMEASIDENTIFIER___Delegate
        , ___FILEBASENAMEASIDENTIFIER___PresenterToInteractorInterface
        , ___FILEBASENAMEASIDENTIFIER___PresenterToViewInterface
        , ___FILEBASENAMEASIDENTIFIER___PresenterToWireframeInterface
        >

@property (nonatomic, strong) ___FILEBASENAMEASIDENTIFIER___Presenter *presenter;

// Test Objects
@property (nonatomic, strong) XCTestExpectation *expectation;

@end

@implementation ___FILEBASENAMEASIDENTIFIER___PresenterTests

- (void)setUp {
        [super setUp];
        
        self.presenter = [___FILEBASENAMEASIDENTIFIER___Presenter new];
        
        self.presenter.interactor = self;
        self.presenter.view = self;
        self.presenter.wireframe = self;
}

- (void)tearDown {
        [super tearDown];
        self.expectation = nil;
}

#pragma mark - Operational

#pragma mark - Interactor to Presenter Interface

#pragma mark - View to Presenter Interface

#pragma mark - Wireframe to Presenter Interface
-(void)testSetDelegateWithAnythingShouldSetPresentersDelegate {
        self.presenter.delegate = self;
        
        XCTAssertEqualObjects (self.presenter.delegate, self);
}

#pragma mark - Presenter to Interactor Interface

#pragma mark - Presenter to View Interface

#pragma mark - Presenter to Wireframe Interface

@end

File: VIPER Tests.xctemplate/Swift/___FILEBASENAME___PresenterTests.swift
import Foundation
import XCTest

@testable import ___PROJECTNAME___

class ___FILEBASENAMEASIDENTIFIER___PresenterTests: XCTestCase {
        var presenter = ___FILEBASENAMEASIDENTIFIER___Presenter()

        // MARK: - Test Objects
        var expectation: XCTestExpectation?

        override func setUp() {
                super.setUp()

                presenter = ___FILEBASENAMEASIDENTIFIER___Presenter()

                presenter.interactor = self
                presenter.view = self
                presenter.wireframe = self
        }

        override func tearDown() {
                super.tearDown()
                expectation = nil
        }

        // MARK: - Operational

        // MARK: - Interactor to Presenter Interface

        // MARK: - View to Presenter Interface

        // MARK: - Wireframe to Presenter Interface
        func testSetDelegateWithAnythingShouldSetPresentersDelegate() {
                // Arrange

                // Act
                presenter.set(delegate: self)

                // Assert
                XCTAssert(presenter.delegate === self)
        }
}

// MARK: Delegate
extension ___FILEBASENAMEASIDENTIFIER___PresenterTests: ___FILEBASENAMEASIDENTIFIER___Delegate {

}

// MARK: - Presenter to Interactor Interface
extension ___FILEBASENAMEASIDENTIFIER___PresenterTests: ___FILEBASENAMEASIDENTIFIER___PresenterToInteractorInterface {

}

// MARK: - Presenter to View Interface
extension ___FILEBASENAMEASIDENTIFIER___PresenterTests: ___FILEBASENAMEASIDENTIFIER___PresenterToViewInterface {

}

// MARK: - Presenter to Wireframe Interface
extension ___FILEBASENAMEASIDENTIFIER___PresenterTests: ___FILEBASENAMEASIDENTIFIER___PresenterToWireframeInterface {

}
